contracts/Utils.sol

```diff
@@ -23,26 +38,36 @@ contract Utils {
 
     mapping(IERC20 => uint256) internal decimals;
 
-    function getUpdateDecimals(IERC20 token) internal returns (uint256) {
-        if (token == ETH_TOKEN_ADDRESS) return ETH_DECIMALS; // save storage access
-        uint256 tokenDecimals = decimals[token];
-        // moreover, very possible that old tokens have decimals 0
-        // these tokens will just have higher gas fees.
-        if (tokenDecimals == 0) {
-            tokenDecimals = token.decimals();
-            decimals[token] = tokenDecimals;
-        }
-
-        return tokenDecimals;
+    function updateDecimals(IERC20 token) internal returns (uint256 tokenDecimals) {
+        tokenDecimals = getDecimals(token);
+        if (
```

Comment:
can call above getConstantDecimals
and if non zero just return reply.

instead of having long if

also if we later add other tokens.
can add in one place. which is better.

hits is kind of a code duplication.

Reply:
Removed duplication